pkg/workflow/management/create.go

```diff
@@ -0,0 +1,37 @@
+package management
+
+import (
+	"github.com/aws/eks-anywhere/pkg/workflow"
+)
+
+const (
+	PreCreateClusterTaskName  workflow.TaskName = "PreCreateManagementCluster"
+	PostCreateClusterTaskName workflow.TaskName = "PostCreateManagementCluster"
+)
+
+// CreateClusterHookRegistrar is a Hook registrar that binds hooks to a create management cluster
+// workflow.
+type CreateClusterHookRegistrar interface {
+	RegisterCreateManagementClusterHooks(workflow.HookBinder)
+}
+
+// CreateClusterConfig defines the configuration for a managment cluster creation workflow.
+type CreateClusterBuilder struct {
+	workflow.HookRegistrarHelper[CreateClusterHookRegistrar]
+}
+
+// Build builds the create cluster workflow.
+func (cfg *CreateClusterBuilder) Build() (*workflow.Workflow, error) {
+	wflw, err := workflow.New(workflow.Config{})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, r := range cfg.HookRegistrars {
```

Comment:
This got me thinking, do we actually need to expose methods to "add" registrars to the builder? If the builder is the entity who knows how to build a workflow and what are the tasks that should be included in it, shouldn't it also know which are right the hook registrars for it?

That would remove the need to have a `WithHookRegistrar` method, which eliminates the debate generics vs code duplication for it.

Maybe we got the point where we need to go and try build one workflow? Maybe the answers would come easier then.

Reply:
I don't think so. The design stipulates an intent to disambiguate the concern of workflows where a workflow should only include tasks that create, upgrade or delete a cluster - nothing more, nothing less. 

Embedding knowledge of registrars here suggests workflows _also_ know about concrete hook capabilities and I think that conflates the scope of a workflow.